### ISSUE TITLE
For a short period of time after a new stable release, beta is an older version.

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#119945]
For instance, as of right now:

```
$ git checkout upstream/beta
Previous HEAD position was 71520442d4 [flutter_releases] Flutter beta 3.4.0-34.0.pre Framework Cherrypicks (#112971)
HEAD is now at 099b3f4bf1 Update Engine revision to 45c5586f2a9583d9511b742703529a92bf8f9db5 for beta release 3.7.0-1.5.pre (#118927)
$ flutter --version 
Downloading Linux x64 Dart SDK from Flutter engine 45c5586f2a9583d9511b742703529a92bf8f9db5...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  238M  100  238M    0     0  87.3M      0  0:00:02  0:00:02 --:--:-- 87.3M
Building flutter tool...
Flutter 3.7.0-1.5.pre • channel unknown • unknown source
Framework • revision 099b3f4bf1 (2 weeks ago) • 2023-01-20 18:35:12 -0800
Engine • revision 45c5586f2a
Tools • Dart 2.19.0 (build 2.19.0-444.6.beta) • DevTools 2.20.1
$ git checkout upstream/stable
Previous HEAD position was 099b3f4bf1 Update Engine revision to 45c5586f2a9583d9511b742703529a92bf8f9db5 for beta release 3.7.0-1.5.pre (#118927)
HEAD is now at 7048ed95a5 Update Engine revision to 800594f1f4a6674010a6f1603c07a919b4d7ebd7 for stable release 3.7.1 (#119687)
$ flutter --version
Downloading Linux x64 Dart SDK from Flutter engine 800594f1f4a6674010a6f1603c07a919b4d7ebd7...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  238M  100  238M    0     0  96.3M      0  0:00:02  0:00:02 --:--:-- 96.4M
Building flutter tool...
Flutter 3.7.1 • channel unknown • unknown source
Framework • revision 7048ed95a5 (2 days ago) • 2023-02-01 09:07:31 -0800
Engine • revision 800594f1f4
Tools • Dart 2.19.1 • DevTools 2.20.1
```

Beta points to an older version of the Dart and Flutter SDK than stable. This is surprising, since I would expect that using the beta channel _always_ gets me a version of Flutter that is >= stable.

One way to solve this would be to update beta to point to the same hash as stable when doing a stable release. We get out of this state once a new beta is cut after a stable.

/cc @zanderso who knows a lot about the release process and might have ideas about why this won't work. Ditto for @christopherfujino 
/cc @CaseyHillers and @XilaiZhang since they're also involved in release.

**Comment by CaseyHillers** [comment]
\cc @itsjustkevin  from the release program side

tl;dr This is intentionally broken. If there's a workflow broken, please let us know so we can prioritize.

Generally, we try to ship a new beta as soon as possible after the stable, but there's some constraints. We will have gone ~2 weeks with this behavior, but we don't expect this to impact any people or tools.

One thing we've proposed is possibly shipping a hidden beta. For example, while 3.7 was on beta in January, we could've created 3.8 beta. While it would've never shipped, it'd be available for us to promote during the stable release, and would live for only ~2 weeks.

**Comment by matanlurey** [comment]
Closing as stale. Please feel free to open a new issue with a strong justification for doing this.

(To be clear, the `-release` and `-infra` teams have no plans to work on this in the known future)
